### DISCUSSION TITLE
[A/L] (2021/22) Relazioni - Esercizio 4

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#276] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152318332-20379546-54e1-40c0-9003-eb02faffda28.png)

**Comment by GattoBrizzolato** [comment]
SI    l'argomento è corretto

**Comment by Elia-Belli** [comment]
No l'argomento non è corretto. Elenchiamo le proprietà di una relazione d'equivalenza $R \subseteq A\times A$:

- **Riflessività**: $\forall a \in A$, aRa
- **Simmetria**: Siano $a,b \in A$, se $aRb \Rightarrow bRa$
- **Transitività**: Siano $a,b,c \in A$, se $aRb \land bRc \Rightarrow aRc$

A differenza della proprietà riflessiva la simmetria non si applica su tutti gli elementi di A, ma agisce solamente in base all'esistenza di un'altra relazione (se esiste aRb allora esiste bRa), quindi non è sempre possibile ricavare la riflessività di tutti gli elementi.

**Controesempio**
Sia R ={(a,b) $\in \mathbb {N}^2$ | a*b >0 }: 2 numeri naturali sono in relazione se il loro prodotto è positivo.

- La simmetria è evidente in quanto la moltiplicazione è un'operazione commutativa: se a*b > 0 allora b*a>0
- La transitività è evidente se a*b>0 e b*c>0 sia a che c sono entrambi positivi allora a*c>0
- C'è però un elemento che non rispetta la riflessività, infatti 0 non è in relazione con se stesso, ma affinchè R sia una relazione di equavalenza tutti gli elementi di $\mathbb N$ devono essere in relazione con stessi, quindi R non è una relazione di equivalenza.

Ecco un caso in cui **valgono simmetria e transitività** ma **non la riflessività**.

**Comment by UnoCasuale** [comment]
Concordo con Elia in quanto possiamo assumere che A sia uguale a {a}.

La funzione sarebbe:
**Transitiva**: a vuoto, non si verificano le ipotesi della transitivita e se le ipotesi sono false la tesi é vera
**Simmetrica** : a vuoto, per lo stesso ragionamento della transitivita 
**Riflessiva**: non lo potrebbe essere un quanto non esisterebbe un b t.c. aRb e bRa
